File: tools/src/main/java/com/yinp/tools/fragment_dialog/CommonDialogFragment.java
package com.yinp.tools.fragment_dialog;

import android.content.Context;
import android.view.View;

import androidx.annotation.LayoutRes;
import androidx.viewbinding.ViewBinding;

/**
 * 公用样式Dialog
 */
public class CommonDialogFragment extends BaseDialogFragment {
    private ViewConvertListener viewConvertListener = null;

    public static CommonDialogFragment newInstance(Context context) {
        return newInstance(context, false);
    }

    public static CommonDialogFragment newInstance(Context context, boolean setKeyListener) {
        CommonDialogFragment dialog = new CommonDialogFragment();
        mContext = context;
        mSetKeyListener = setKeyListener;
        return dialog;
    }

    /**
     * 设置Dialog布局
     *
     * @param layoutId
     * @return
     */
    public CommonDialogFragment setLayoutId(@LayoutRes int layoutId) {
        this.mLayoutResId = layoutId;
        return this;
    }

    public CommonDialogFragment setLayout(ViewBinding viewBinding) {
        this.binding = viewBinding;
        this.layoutView = viewBinding.getRoot();
        return this;
    }

    /**
     * 用于通讯
     *
     * @param convertListener
     * @return
     */
    public CommonDialogFragment setViewConvertListener(ViewConvertListener convertListener) {
        viewConvertListener = convertListener;
        return this;
    }

    @Override
    protected int updateLayoutId() {
        return mLayoutResId;
    }

    @Override
    protected View updateLayoutView() {
        return layoutView;
    }

    /**
     * 通过id处理各种事件
     *
     * @param holder
     * @param dialog
     */
    @Override
    public void convertView(DialogFragmentHolder holder, BaseDialogFragment dialog) {
        if (viewConvertListener != null) {
            viewConvertListener.convertView(holder, dialog);
        }
    }

    @Override
    protected void convertView(DialogFragmentHolder holder, BaseDialogFragment dialog, ViewBinding viewBinding) {
        if (viewConvertListener != null) {
            viewConvertListener.convertView(holder, dialog, viewBinding);
        }
    }
}

File: tools/src/main/java/com/yinp/tools/fragment_dialog/CommonDialogFragment.kt
package com.yinp.tools.fragment_dialog

import android.view.View
import androidx.annotation.LayoutRes
import androidx.viewbinding.ViewBinding

/**
 * author:yinp
 * 2022/4/24+21:18
 * describe:
 */
class CommonDialogFragment(var setKeyListener: Boolean = false) : BaseDialogFragment(setKeyListener) {
    private var viewConvertListener: ViewConvertListener? = null

    companion object {
        fun newInstance(): CommonDialogFragment {
            return CommonDialogFragment()
        }
    }

    /**
     * 设置Dialog布局
     *
     * @param layoutId
     * @return
     */
    fun setLayoutId(@LayoutRes layoutId: Int): CommonDialogFragment {
        mLayoutResId = layoutId
        return this
    }

    fun setLayout(viewBinding: ViewBinding): CommonDialogFragment {
        binding = viewBinding
        layoutView = viewBinding.root
        return this
    }

    /**
     * 用于通讯
     *
     * @param convertListener
     * @return
     */
    fun setViewConvertListener(convertListener: ViewConvertListener?): CommonDialogFragment {
        viewConvertListener = convertListener
        return this
    }

    override fun updateLayoutId(): Int {
        return mLayoutResId
    }

    override fun updateLayoutView(): View? {
        return layoutView
    }

    /**
     * 通过id处理各种事件
     *
     * @param holder
     * @param dialog
     */
    override fun convertView(holder: DialogFragmentHolder, dialog: BaseDialogFragment) {
        if (viewConvertListener != null) {
            viewConvertListener!!.convertView(holder, dialog)
        }
    }

    override fun convertView(
        holder: DialogFragmentHolder,
        dialog: BaseDialogFragment,
        viewBinding: ViewBinding
    ) {
        if (viewConvertListener != null) {
            viewConvertListener!!.convertView(holder, dialog, viewBinding)
        }
    }
}